chainer/gradient_check.py

```diff
@@ -583,28 +590,30 @@ def _directional_numeric_gradients(self, directions, y0_data):
         no_grads = self.no_grads
         dtype = self.dtype
         detect_nondifferentiable = self.detect_nondifferentiable
+        params_data = [p if self.params_to_func else p.array for p in params]
 
         xp = device.xp
 
         x_vars = [variable.Variable(x, requires_grad=False) for x in x_data]
-        variables = (
-            [x for x, no_grad in six.moves.zip(x_vars, no_grads)
-             if not no_grad]
-            + list(params))
+
+        x_data_filtered = [
+            x.array for x, skip in six.moves.zip(x_vars, no_grads) if not skip]
 
         if dtype is None:
-            casted_data = [x.array for x in variables]
+            casted_data = [x for x in x_data_filtered + params_data]
```

Comment:
`casted_data` has both `ndarray`s and `Parameter`s. That looks weird. `data` stands for "array".

Reply:
Doesn't `casted_data` only contains "arrays" in the current implementation?